src/plugins/internalnotification/internalnotification.c

```diff
@@ -241,6 +243,70 @@ static int keySetContainsSameOrBelow (Key * check, KeySet * ks)
 	return 0;
 }
 
+void elektraInternalnotificationNotifyChangedKeys (Plugin * plugin, const KeySetDiff * diff)
+{
+	PluginState * pluginState = elektraPluginGetData (plugin);
+	ELEKTRA_ASSERT (pluginState != NULL, "plugin state was not initialized properly");
+
+	KeySet * modifiedKeys = elektraChangeTrackingGetModifiedKeys (diff);
+	// TODO: do we also consider added and removed keys as modified here?
```

Comment:
This is something that I'm not sure yet.

Reply:
Don't the unit tests check this? Unit tests might be disabled. If not, please add some. Intuitively, both should trigger a notification, as the added or removed key might be a key the user is subscribed to? (Removing would go back to default once the spec plugin works properly.)